fix(ContentTypeCache): localContentType is converted to HubContentType (fixes #308)

JPSchellenberg opened the pull request:
Hey,

as discussed in #308 I fixed the problem, that a localContentType was not properly converted to a HubContentType and therefore was lacking the `.canBeInstalledBy`-method. This happened if a ContentTypeCache was loaded from storage.

JPSchellenberg committed:
fix(ContentTypeCache): localContentType is converted to HubContentType

sr258 commented on src/ContentTypeCache.ts:
```diff
@@ -172,11 +172,15 @@ export default class ContentTypeCache {
         if (!machineNames || machineNames.length === 0) {
             return cache;
```
Won't we need to make sure that the function is attached if all contentTypes are requested? (The return value you've changed below is only used in certain cases if you look for a specific contentType).

JPSchellenberg replied:
You are completely right. My bad. I will change this today.

JPSchellenberg committed:
fix(ContentTypeCache): convert all HubContentTypes - not only filtered

JPSchellenberg committed:
test(ContentTypeCache): returned HubContentTypes have .canBeInstalledBy

sr258 committed:
test(ContenTypeCache): corrected test: returned HubContentTypes have .canBeInstalledBy

sr258 reviewed (dismissed):
The test didn't work out the way it was supposed to, as the cache is not part of the EditorConfig but stored aside it. It also doesn't store the raw format returned from the HTTP endpoint but a transformed version. I've changed the test accordingly.

JPSchellenberg committed:
fix(ContentTypeCache): convert only cache entries loaded from storage

cache entries loaded via .forceUpdate() do not need to be converted as they are already converted within .forceUpdate()

JPSchellenberg committed:
convert HubContentType to interface

JPSchellenberg committed:
use .canInstallLibrary instead of .canBeInstalledBy

JPSchellenberg committed:
test(ContentTypeCache): remove .canBeInstalledBy test

JPSchellenberg committed:
fix(ContentTypeCache): remove deprecated conversion to LocalFormat
